p11-kit/delete-profile.c

```diff
@@ -115,27 +117,11 @@ delete_profile (const char *token_str,
 		goto cleanup;
 	}
 
+	/* Module and session should always set at this point.  */
 	module = p11_kit_iter_get_module (iter);
-	if (module == NULL) {
-		p11_message (_("failed to obtain module"));
-		goto cleanup;
-	}
-	slot = p11_kit_iter_get_slot (iter);
-
-	rv = module->C_OpenSession (slot, CKF_SERIAL_SESSION | CKF_RW_SESSION, NULL, NULL, &session);
-	if (rv != CKR_OK) {
-		p11_message (_("failed to open session: %s"), p11_kit_strerror (rv));
-		goto cleanup;
-	}
-
-	pin = p11_kit_uri_get_pin_value (uri);
-	if (pin != NULL) {
-		rv = module->C_Login (session, CKU_USER, (unsigned char *)pin, strlen (pin));
-		if (rv != CKR_OK) {
-			p11_message (_("failed to login: %s"), p11_kit_strerror (rv));
-			goto cleanup;
-		}
-	}
+	return_val_if_fail (module, 1);
```

Comment:
nit: I would prefer to have all of these specify `!= NULL` so its consistent with the rest of the code. But it doesnt really matter